python/beeai_framework/agents/react/runners/default/runner.py

```diff
@@ -325,13 +325,15 @@ async def _init_memory(self, input: ReActAgentRunInput) -> BaseMemory:
             ]
         )
 
-        if memory.is_empty() or input.prompt:
-            created_at = datetime.datetime.now(tz=datetime.UTC)
-            content = (
-                self.templates.user.render(UserPromptTemplateInput(input=input.prompt, created_at=created_at))
-                if input.prompt
-                else self.templates.user_empty.render(UserEmptyPromptTemplateInput())
-            )
+        created_at = datetime.datetime.now(tz=datetime.UTC)
+        if input.prompt:
+            last_msg = input.prompt if isinstance(input.prompt, str) else input.prompt[-1].text
+            content = self.templates.user.render(UserPromptTemplateInput(input=last_msg, created_at=created_at))
+            if isinstance(input.prompt, list):
+                await memory.add_many(input.prompt[:-1])
+            await memory.add(UserMessage(content=content, meta={"createdAt": created_at}))
+        elif memory.is_empty():
```

Comment:
1. It seems that `memory.is_empty()` is always false, because it contains the System message. I would change it to `if len(memory.messages) < 2` (`elif` not needed)
2. `last_msg = input.prompt if isinstance(input.prompt, str) else input.prompt[-1].text` should only apply if the input is `str`, if it is a `list`, just append all messages and don't change anything